.github/workflows/ci.yaml

```diff
@@ -3,9 +3,6 @@ name: Fmt and Clippy
 # This job runs rustfmt and clippy linting,
 
 on:
-  push:
-    branches: main
```

Comment:
We should **never** push to main directly. This is just a waste of money and time (It may take up time from actual actions to be executed)

Reply:
I believe this will also trigger when merging a PR to main, it isn't only on literal pushes to main. It can still be valuable to test on main if the PR was not up to date, but if the PR is up to date then it should indeed be wasteful.

Reply:
It is indeed also PR merge to main.

I just do not understand when it should diverge from the branch.

If what is on main is out of date but correctly formatted and clippy provides no error, and so is the source branch, I believe there would be a mistake in the merging of git if that should diverge when merged to main? Alternatively the merging should provide a conflict resulting in no merge?

Same argument goes for testing.

Reply:
I don't know if there is an easy way to enable that rule across all of our repositories but that might be nice if there is such a way.

Reply:
We also had a couple of edge cases where the CI pipeline was green on the pull request but failed once merged into main due to the time passed.

One edge case, I remember was when the CI pipeline pulled e.g. the latest docker image and this image was updated by someone in between the pull request CI pipeline run and the main CI pipeline run.

Reply:
You can read more about this situation here: https://bors.tech/essay/2017/02/02/pitch/

Bors is a GitHub bot that secures against this but GitHub has since introduced native settings to secure against it so using Bors is not required any more I think.

Reply:
Sure. I will remove this line from the PR.

Reply:
Or I just enabled above check

Reply:
![image](https://github.com/user-attachments/assets/92c45a37-60d5-4b18-964f-a19bf8909780)

Reply:
How costly is it to run these things on main? Surely it isn't that much? I like that setting and think it should be enabled but I don't really see the huge benefit of disabling main checks.

Reply:
Sure. Around a dollar a month, the whole issue is cost and that it takes up time. So a really small issue yes.